src/app/icd/ICDData.h

```diff
@@ -0,0 +1,127 @@
+// Set to 1 to enforce SIT Slow Polling Max value to 15seconds (spec 9.16.1.5)
+#define ICD_ENFORCE_SIT_SLOW_POLL_LIMIT 0
+#endif
+
+namespace chip {
+
+namespace app {
+// Forward declaration of ICDManager to allow it to be friend with ICDData
+class ICDManager;
+} // namespace app
+
+/**
+ * @brief ICDData manages and stores ICD related configurations for the ICDManager.
+ *        Goal of the class is to expose ICD information to all consummers without creating circular dependencies
+ *        since the ICDManager is appart of the App layer
+ *
+ *        Anyone can read the ICD configurations but only the ICDManager can changes those configurations.
+ */
+class ICDData
+{
+public:
+    enum class ICDMode : uint8_t
+    {
+        SIT, // Short Interval Time ICD
+        LIT, // Long Interval Time ICD
+    };
+
+    static ICDData & GetInstance() { return instance; };
+
+    uint32_t GetIdleModeDurationSec() { return mIdleInterval_s; }
+
+    uint32_t GetActiveModeDurationMs() { return mActiveInterval_ms; }
+
+    uint16_t GetActiveModeThresholdMs() { return mActiveThreshold_ms; }
+
+    uint32_t GetICDCounter() { return mICDCounter; }
+
+    uint16_t GetClientsSupportedPerFabric() { return mFabricClientsSupported; }
+
+    System::Clock::Milliseconds32 GetSITPollingThreshold() { return kSITPollingThreshold; }
+
+    System::Clock::Milliseconds32 GetFastPollingInterval() { return mFastPollingInterval; }
+
+    /**
+     * If ICD_ENFORCE_SIT_SLOW_POLL_LIMIT is set to 0, function will always return the configured Slow Polling interval
+     * (CHIP_DEVICE_CONFIG_ICD_SLOW_POLL_INTERVAL).
+     *
+     * If ICD_ENFORCE_SIT_SLOW_POLL_LIMIT is set to 1, the returned value will depend on the devices operating mode.
+     * If ICDMode == SIT && the configured slow poll interval is superior to the maximum threshold (15s), the function will return
+     * the threshold (15s). If ICDMode == SIT but the configured slow poll interval is equal or inferior to the threshold, the
+     * function will the return the configured slow poll interval. If ICDMode == LIT, the function will return the configured slow
+     * poll interval.
+     *
+     * @return System::Clock::Milliseconds32
+     */
+    System::Clock::Milliseconds32 GetSlowPollingInterval();
+
+    ICDMode GetICDMode() { return mICDMode; }
+
+private:
+    // Singleton Object
+    ICDData() = default;
+    static ICDData instance;
+
+    // ICD related information is managed by the ICDManager but stored in the ICDData to enable consummers to access it without
+    // creating a circular dependency. To avoid allowing consummers changing the state of the ICD mode without going through the
+    // ICDManager, the ICDManager is a friend that can access the private setters. If a consummer needs to be notified when a value
+    // is changed, they leverages the Observer events the ICDManager generates. See src/app/icd/ICDStateObserver.h
+    friend class chip::app::ICDManager;
```

Comment:
Doesn't this cause a dependency on ICDManager that users of ICDData don't want ?

Reply:
No it doesn't it; Same logic as adding test classes as friends when they are not always compiled. If the class isn't present, this is ignored.